[SecurityBundle][Security][Finder] Remove deprecated code paths

ogizanagi opened the pull request:
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | yes
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 1b600b0c94cf3a7eb65e847e8d9a19de7bfa8514, 366aefd75f71d5a35647634d91259932cfa39907, 5b7fe852aa1ce7e8fb01c8ed567af05b288b5d43
| License       | MIT
| Doc PR        | N/A

Removing my own deprecated stuff from 4.0 :)

ogizanagi committed:
[SecurityBundle][Security][Finder] Remove deprecated code paths

- [Finder] Removed `ExceptionInterface`
- [SecurityBundle] remove `UserPasswordEncoderCommand` BC layer
- [Security] remove `LogoutUrlGenerator::registerListener` BC layer

nicolas-grekas commented:
rebase needed to see tests green

ogizanagi commented:
Aaaand, green it is.

ogizanagi commented on src/Symfony/Component/Security/Http/Logout/LogoutUrlGenerator.php:
```diff
@@ -48,21 +48,8 @@ public function __construct(RequestStack $requestStack = null, UrlGeneratorInter
      * @param string|null                    $context          The listener context
      */
-    public function registerListener($key, $logoutPath, $csrfTokenId, $csrfParameter, CsrfTokenManagerInterface $csrfTokenManager = null/*, $context = null*/)
+    public function registerListener($key, $logoutPath, $csrfTokenId, $csrfParameter, CsrfTokenManagerInterface $csrfTokenManager = null, string $context = null)
```
Related PR to add the typehint in 3.3: https://github.com/symfony/symfony/pull/22757

nicolas-grekas commented:
could the commit message be made more explicit? any CHANGELOG missing an update?

ogizanagi commented:
I changed the commit message and description and added an entry in the `Finder` component's CHANGELOG to mention the removal of the `ExceptionInterface`. Regarding the security é security bundle changelogs, I don't now if an entry is really necessary nor how to turn it right.
